backends/bmv2/CMakeLists.txt

```diff
@@ -118,7 +118,12 @@ else()
 endif()
 
 set (GTEST_BMV2_SOURCES
-  ${CMAKE_SOURCE_DIR}/test/gtest/bmv2_isvalid.cpp
+  ${P4C_SOURCE_DIR}/test/gtest/bmv2_isvalid.cpp
+
+  # XXX(seth): This should not depend on BMV2. The source of the dependency is
+  # the SynthesizeValidField pass; we should either promote it to a midend pass
```

Comment:
That pass is responsible for issue #841, so it cannot be moved to the mid-end, since it breaks some important invariants about the IR.

Reply:
I know, yeah. It's on my to-do list to deal with that better. I have a plan, but I haven't implemented it yet. (I want to try it on our proprietary backend first, basically.)